### PR TITLE
Patricia dev

### PR DESCRIPTION
**Pull request by awindler**

### CODE REVIEW

File: oscillogram_classification/cam.py [hunk 1]
```diff
@@ -228,23 +228,25 @@ def generate_hirescam(input_array: np.ndarray, trained_model: keras.models.Model
...
     :param title: window title, e.g., recorded vehicle component and classification result
+    :param time_vals: time values to be visualized on the x-axis
     """
     plt.rcParams["figure.figsize"] = len(cams) * 10, 3
     fig, axes = plt.subplots(nrows=1, ncols=len(cams), sharex=True, sharey=True)
     fig.canvas.set_window_title(title)
     # bounding box in data coordinates that the image will fill (left, right, bottom, top)
-    extent = [0, cams[list(cams.keys())[0]].shape[0], np.floor(np.min(voltage_vals)), np.ceil(np.max(voltage_vals))]
-    data_points = np.array([i for i in range(len(voltage_vals))])
+    extent = [0, np.max(time_vals), np.floor(np.min(voltage_vals)), np.ceil(np.max(voltage_vals))]
```
Comment (tbohne) [inline oscillogram_classification/cam.py]: A little wasteful in terms of computational resources to use `np.max` here, we know it's always the last value, so `time_vals[-1]` is better.

File: oscillogram_classification/cam.py [hunk 2]
```diff
@@ -276,33 +279,35 @@ def gen_heatmaps_as_overlay(cams: dict, voltage_vals: np.ndarray, title: str) ->
...
     :param title: window title, e.g., recorded vehicle component and classification result
+    :param time_vals: time values to be visualized on the x-axis
     """
     plt.rcParams["figure.figsize"] = len(cams) * 10, 4
     fig, axes = plt.subplots(nrows=2, ncols=len(cams), sharex=True, sharey=True)
     fig.canvas.set_window_title(title)
     # bounding box in data coordinates that the image will fill (left, right, bottom, top)
-    extent = [0, cams[list(cams.keys())[0]].shape[0], np.floor(np.min(voltage_vals)), np.ceil(np.max(voltage_vals))]
-    data_points = [i for i in range(len(voltage_vals))]
+    extent = [0, np.max(time_vals), np.floor(np.min(voltage_vals)), np.ceil(np.max(voltage_vals))]
```
Comment (tbohne) [inline oscillogram_classification/cam.py]: Same here.

File: oscillogram_classification/cam.py [hunk 3]
```diff
@@ -254,20 +256,21 @@ def gen_heatmaps_overlay_side_by_side(cams: dict, voltage_vals: np.ndarray, titl
             cams[list(cams.keys())[i]][np.newaxis, :], cmap="plasma", aspect="auto", alpha=.75, extent=extent
         )
         # data
-        axes[i].plot(data_points, voltage_vals, '#000000')
+        axes[i].plot(time_vals, voltage_vals, '#000000')
     plt.tight_layout()
 
 
-def gen_heatmaps_as_overlay(cams: dict, voltage_vals: np.ndarray, title: str) -> Image:
+def gen_heatmaps_as_overlay(cams: dict, voltage_vals: np.ndarray, title: str, time_vals: List[float]) -> Image:
```
Comment (tbohne) [inline oscillogram_classification/cam.py:263]: It's generally not a good idea to change the signature of functions that are used in the other repositories, it will stop being compatible with all the usages. In this case, since it probably makes sense to always return these plots with seconds on the y-axis, I tend to still accept it and change the function calls in `vehicle_diag_smach`. Already opened an issue for this.

Comment (tbohne) [reply oscillogram_classification/cam.py:263]: Better option would be to do just as you did in the preprocessing below, but as said, in this case it is fine.

File: oscillogram_classification/preprocess.py [hunk 4]
```diff
@@ -38,7 +39,11 @@ def read_oscilloscope_recording(rec_file: str) -> Tuple[int, List[float]]:
     df = pd.read_csv(rec_file, delimiter=';', na_values=['-∞', '∞'])
     df = df[1:].apply(lambda x: x.str.replace(',', '.')).astype(float).dropna()
     curr_voltages = df['Kanal A'].to_list()
-    return label, curr_voltages
+    if not return_time_values:
+        return label, curr_voltages
+    else:
```
Comment (tbohne) [inline oscillogram_classification/preprocess.py]: `else` can be removed

**Review by tbohne — CHANGES_REQUESTED** [review]

**Review by tbohne — APPROVED** [review]
